org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectRegistryManager.java

```diff
@@ -740,56 +739,66 @@ private Map<IFile, MavenProjectFacade> readMavenProjectFacades(Collection<IFile>
     Map<IFile, MavenProjectFacade> result = new HashMap<>(poms.size(), 1.f);
     for(Entry<ResolverConfiguration, Collection<IFile>> entry : groupsToImport.entrySet()) {
       ResolverConfiguration resolverConfiguration = entry.getKey();
-      Collection<IFile> pomFiles = entry.getValue();
-      result.putAll(execute(state, poms.size() == 1 ? pomFiles.iterator().next() : null, resolverConfiguration,
-          (executionContext, pm) -> {
-            Map<File, MavenExecutionResult> mavenResults = getMaven().readMavenProjects(pomFiles.stream()
-                .filter(IFile::isAccessible).map(ProjectRegistryManager::toJavaIoFile).collect(Collectors.toList()),
-                executionContext.newProjectBuildingRequest());
-
-            Map<IFile, MavenProjectFacade> facades = new HashMap<>(mavenResults.size(), 1.f);
-            for(IFile pom : pomFiles) {
-              if(!pom.isAccessible()) {
-                continue;
-              }
-              MavenExecutionResult mavenResult = mavenResults.get(ProjectRegistryManager.toJavaIoFile(pom));
-              MavenProject mavenProject = mavenResult.getProject();
-              MarkerUtils.addEditorHintMarkers(markerManager, pom, mavenProject, IMavenConstants.MARKER_POM_LOADING_ID);
-              markerManager.addMarkers(pom, IMavenConstants.MARKER_POM_LOADING_ID, mavenResult);
-              if(mavenProject != null && mavenProject.getArtifact() != null) {
-                MavenProjectFacade mavenProjectFacade = new MavenProjectFacade(ProjectRegistryManager.this, pom,
-                    mavenProject, resolverConfiguration);
-                putMavenProject(mavenProjectFacade, mavenProject); // maintain maven project cache
-                facades.put(pom, mavenProjectFacade);
-              }
+      Map<PlexusContainer, List<IFile>> pomFiles = mapToContainer(entry.getValue());
+      SubMonitor containerMonitor = subMonitor.split(pomFiles.size());
+      containerMonitor.setWorkRemaining(pomFiles.size());
+      for(var containerEntry : pomFiles.entrySet()) {
+        List<IFile> fileList = containerEntry.getValue();
+        MavenExecutionContext context = new MavenExecutionContext(containerEntry.getKey(), null);
+        configureExecutionRequest(context.getExecutionRequest(), state, fileList.size() == 1 ? fileList.get(0) : null,
+            resolverConfiguration);
+
+        result.putAll(context.execute((ctx, mon) -> {
```

Comment:
Looks like you forgot to set the execution result into the 'result' map.
Lets see if that fixes the tests.